libcxx/test/libcxx/selftest/sh.cpp/run-success.sh.cpp

```diff
@@ -8,4 +8,4 @@
 
 // Make sure the test passes if it succeeds to run
 
-// RUN: true
+// RUN: :
```

Comment:
I kinda wish there was a Lit builtin command called `true` (and why not `false`) since this would be a lot more readable, but this LGTM.

CC @jdenny-ornl

Reply:
Why are true/false not available in the windows environment that motivated this patch?  I'm not a windows user, but I don't recall hearing of this issue before.

Reply:
Lit's test suite has [tests where RUN lines containing true/false execute with lit's internal shell](https://github.com/llvm/llvm-project/blob/main/llvm/utils/lit/tests/Inputs/shtest-run-at-line/internal-shell/basic.txt) (not bash), and they [appear to succeed in windows bots](https://lab.llvm.org/buildbot/#/builders/216).  That suggests to me that true/false are available as external commands there.  Maybe @dyung can shed some light on that setup.

Is the goal of this patch to relax windows installation requirements, or am I just misunderstanding what's happening?

Reply:
My goal is just to ensure that we use Lit builtin commands exclusively whenever possible so as to decouple our test suite from the underlying platform as much as possible. `true` is not a Lit builtin but we have a Lit builtin equivalent (`:`), so this patch seems desirable. I'll go ahead and merge this but I agree it'd be interesting to figure out how these tests manage to work on Windows.

Reply:
> Lit's test suite has [tests where RUN lines containing true/false execute with lit's internal shell](https://github.com/llvm/llvm-project/blob/main/llvm/utils/lit/tests/Inputs/shtest-run-at-line/internal-shell/basic.txt) (not bash), and they [appear to succeed in windows bots](https://lab.llvm.org/buildbot/#/builders/216). That suggests to me that true/false are available as external commands there. Maybe @dyung can shed some light on that setup.
> 
> Is the goal of this patch to relax windows installation requirements, or am I just misunderstanding what's happening?

I'm not completely sure what is being asked here, but I took a look at our Windows bot that @jdenny-ornl linked to and while there is a true.exe that seems to be installed with git (C:\Program Files\Git\usr\bin), but unless something is adding that directory to the path during testing, it is NOT a part of the regular path. And indeed, trying to run "true" from a command prompt gives the expected error `'true' is not recognized as an internal or external command, operable program or batch file.`.

Reply:
@dyung Thanks.  That's exactly what I was looking for.  It looks like [lit is adding that directory here](https://github.com/llvm/llvm-project/blob/1d8985fc14457429c8120b09ae842250bdd225b1/llvm/utils/lit/lit/llvm/config.py#L46) based on the following [output from that windows bot](https://lab.llvm.org/buildbot/#/builders/216/builds/29020/steps/7/logs/stdio):

```
llvm-lit.py: Z:\b\llvm-clang-x86_64-sie-win\llvm-project\llvm\utils\lit\lit\llvm\config.py:46: note: using lit tools: C:\Program Files\Git\usr\bin
```